auth/src/main/java/com/firebase/ui/auth/data/model/NetworkException.java

```diff
@@ -0,0 +1,15 @@
+package com.firebase.ui.auth.data.model;
+
+import android.support.annotation.NonNull;
+
+import com.firebase.ui.auth.ErrorCodes;
+import com.google.firebase.auth.FirebaseAuthException;
+
+/**
+ * Represents an error connecting to the internet.
+ */
+public class NetworkException extends FirebaseAuthException {
```

Comment:
I'm not sure if we want to extend from Firebase auth exceptions or just do our own.

Reply:
We should do our own, see comment above about how it might look.  Extending `FirebaseAuthException` doesn't really get us anything and is strange if we're propagating errors that might come from within our framework or a non-Firebase SDK.